libraries/base/Java/Primitive.hs

```diff
@@ -57,6 +62,22 @@ instance Data Short where
 
 deriving instance Typeable JChar
 
+charToJChar :: Char -> Maybe JChar
+charToJChar chr
+    | minJChar <= chrVal && chrVal <= maxJChar = Just $ fromIntegral chrVal
+    | otherwise = Nothing
+    where minJChar = fromIntegral (minBound :: JChar)
+          maxJChar = fromIntegral (maxBound :: JChar)
+          chrVal = ord chr
```

Comment:
You should consider surrogates (mentioned below) here too. The output `JChar` should *not* be a surrogate.

Reply:
@rahulmutt Thanks, appreciate this. Do you think that I should drop the range constraints from JChar? I thought about the fact that it holds trivially under the current definition of JChar but I added them in case it changes in the future. I will also take a look at the surrogates and submit a new PR.